src/eradius_server.erl

```diff
@@ -92,27 +91,30 @@
 -callback radius_request(#radius_request{}, #nas_prop{}, HandlerData :: term()) -> 
     {reply, #radius_request{}} | noreply | {error, timeout}.
 
-%% @private
 -spec start_link(atom(), inet:ip4_address(), port_number()) -> {ok, pid()} | {error, term()}.
-start_link(ServerName, IP = {A,B,C,D}, Port) ->
+start_link(ServerName, IP, Port) ->
+    start_link(ServerName, IP, Port, []).
+
+-spec start_link(atom(), inet:ip4_address(), port_number(), [inet:socket_setopt()]) -> {ok, pid()} | {error, term()}.
+start_link(ServerName, IP = {A,B,C,D}, Port, Opts) ->
     Name = list_to_atom(lists:flatten(io_lib:format("eradius_server_~b.~b.~b.~b:~b", [A,B,C,D,Port]))),
-    gen_server:start_link({local, Name}, ?MODULE, {ServerName, IP, Port}, []).
+    gen_server:start_link({local, Name}, ?MODULE, {ServerName, IP, Port, Opts}, []).
 
 stats(Server, Function) ->
     gen_server:call(Server, {stats, Function}).
 
 %% ------------------------------------------------------------------------------------------
 %% -- gen_server Callbacks
 %% @private
-init({ServerName, IP, Port}) ->
+init({ServerName, IP, Port, Opts}) ->
     process_flag(trap_exit, true),
     RecBuf = application:get_env(eradius, recbuf, 8192),
-    case gen_udp:open(Port, [{active, once}, {ip, IP}, binary, {recbuf, RecBuf}]) of
+    case gen_udp:open(Port, [{active, once}, {ip, IP}, binary, {recbuf, RecBuf}] ++ Opts) of
         {ok, Socket} ->
             {ok, #state{socket = Socket,
-                        ip = IP, port = Port, name = ServerName,
-                        transacts = ets:new(transacts, []),
-                        counter = eradius_counter:init_counter({IP, Port, ServerName})}};
+                ip = IP, port = Port, name = ServerName,
```

Comment:
Please add back spaces as was before to align record fields.

Reply:
done

Reply:
Thanks!